service-worker/simple-service-worker/app.js

```diff
@@ -42,7 +42,7 @@ const createGalleryFigure = async (galleryImage) => {
     const myImage = document.createElement('img');
     const myCaption = document.createElement('caption');
     const myFigure = document.createElement('figure');
-    myCaption.textContent = `${galleryImage.name}: Taken by ${galleryImage.credit}`;
+    myCaption.innerHTML = `${galleryImage.name}: Taken by ${galleryImage.credit}`;
```

Comment:
Even though this works, we need to avoid [XSS](https://developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting). Instead of assigning the unescaped `galleryImage.name` via `innerHTML`, it would be better to split this up into two `span` elements, one of which is assigned via `textContent` and the other via `innerHTML`:

```suggestion
    const myName = document.createElement('span');
    myName.textContent = `${galleryImage.name}: `;
    const myCredit = document.createElement('span');
    myCredit.innerHTML = `Taken by ${galleryImage.credit}`;
    myCaption.append(myName, myCredit);
```

Ideally, the "Taken by " prefix would even move into the data.